test/ngRoute/routeSpec.js

```diff
@@ -1677,62 +1678,380 @@ describe('$route', function() {
+          expect(routeUpdate).not.toHaveBeenCalled();
+          expect($routeParams).toEqual({});
+        });
+      }
+    );
+
+
+    it('should not reload a route when `reloadOnUrl` is disabled and URL maps to the same route',
+      function() {
+        var routeChange = jasmine.createSpy('routeChange');
+        var routeUpdate = jasmine.createSpy('routeUpdate');
+
+        module(function($routeProvider) {
+          $routeProvider.when('/path/:param', {reloadOnUrl: false});
+        });
+
+        inject(function($location, $rootScope, $routeParams) {
+          $rootScope.$on('$routeChangeStart', routeChange);
+          $rootScope.$on('$routeChangeSuccess', routeChange);
+          $rootScope.$on('$routeUpdate', routeUpdate);
+
+          expect(routeChange).not.toHaveBeenCalled();
+
+          // Initial load
+          $location.path('/path/foo');
+          $rootScope.$digest();
+          expect(routeChange).toHaveBeenCalledTimes(2);
+          expect(routeUpdate).not.toHaveBeenCalled();
+          expect($routeParams).toEqual({param: 'foo'});
+
+          routeChange.calls.reset();
+
+          // Route update (no reload)
+          $location.path('/path/bar').search('foo', 'bar').hash('baz');
+          $rootScope.$digest();
+          expect(routeChange).not.toHaveBeenCalled();
+          expect(routeUpdate).toHaveBeenCalledOnce();
+          expect($routeParams).toEqual({param: 'bar', foo: 'bar'});
+        });
+      }
+    );
+
+
+    it('should update `$routeParams` even when not reloading a route', function() {
+      var routeChange = jasmine.createSpy('routeChange');
 
       module(function($routeProvider) {
-        $routeProvider.when('/foo', {controller: angular.noop, reloadOnSearch: false});
+        $routeProvider.when('/path/:param', {reloadOnUrl: false});
       });
 
-      inject(function($route, $location, $rootScope) {
+      inject(function($location, $rootScope, $routeParams) {
         $rootScope.$on('$routeChangeStart', routeChange);
         $rootScope.$on('$routeChangeSuccess', routeChange);
-        $rootScope.$on('$routeUpdate', routeUpdate);
 
         expect(routeChange).not.toHaveBeenCalled();
 
-        $location.path('/foo');
+        // Initial load
+        $location.path('/path/foo');
         $rootScope.$digest();
-        expect(routeChange).toHaveBeenCalled();
         expect(routeChange).toHaveBeenCalledTimes(2);
-        expect(routeUpdate).not.toHaveBeenCalled();
+        expect($routeParams).toEqual({param: 'foo'});
+
         routeChange.calls.reset();
 
-        // don't trigger reload
-        $location.search({foo: 'bar'});
+        // Route update (no reload)
+        $location.path('/path/bar');
         $rootScope.$digest();
         expect(routeChange).not.toHaveBeenCalled();
-        expect(routeUpdate).toHaveBeenCalled();
+        expect($routeParams).toEqual({param: 'bar'});
       });
     });
 
 
+    describe('reload', function() {
```

Comment:
Can this describe be a bit more explicit? There's already a top-level describe with this name. How about `with reload()`

Reply:
I don't see a top-level `describe` with the same name 😕.

But I agree the descriptions are not great (I copied them from the existing `reloadOnSearch` tests 😳).
I've updated both to ``with `$route.reload()` `` (since they are testing how the configs work in conjuction with `$route.reload()`).

Reply:
Yeah, I didn't see the reload() block was in reloadOnSearch